dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataset/hibernate/HibernateSectionStore.java

```diff
@@ -63,4 +64,16 @@ public Section getSectionByName( String name, DataSet dataSet )
             .addPredicate( root -> builder.equal( root.get( "name" ), name ) )
             .addPredicate( root -> builder.equal( root.get( "dataSet" ), dataSet ) ) );
     }
+
+    @Override
+    public List<Section> getSectionsByDataElement( String dataElementUid )
+    {
+        String hql = "select * from section s"
+            + " left join sectiondataelements sde on s.sectionid = sde.sectionid"
+            + " left join sectiongreyedfields sgf on s.sectionid = sgf.sectionid"
+            + " left join dataelementoperand deo on sgf.dataelementoperandid = deo.dataelementoperandid"
+            + ", dataelement de"
+            + " where de.uid = :Id and (sde.dataelementid = de.dataelementid or deo.dataelementid = de.dataelementid);";
+        return getSession().createNativeQuery( hql, Section.class ).setParameter( "Id", dataElementUid ).list();
```

Comment:
Perhaps use `:dataElementId` as parameter just to make it explicit what it refers to and prepare for additional parameters later.

Reply:
done